NDTensors/ext/NDTensorsCUDAExt/mul.jl

```diff
@@ -0,0 +1,27 @@
+# This was calling generic matrix multiplication.
+# TODO: Raise an issue with `CUDA.jl`.
+function mul!(
+  CM::Exposed{<:CuArray,<:LinearAlgebra.Transpose},
+  AM::Exposed{<:CuArray},
+  BM::Exposed{<:CuArray},
+  α,
+  β,
+)
+  return mul!(parent(CM), transpose(BM), transpose(AM), α, β)
+
+  return unexpose(CM)
+end
+
+# This was calling generic matrix multiplication.
+# TODO: Raise an issue with `CUDA.jl`.
+function mul!(
+  CM::Exposed{<:CuArray,<:LinearAlgebra.Adjoint},
+  AM::Exposed{<:CuArray},
+  BM::Exposed{<:CuArray},
+  α,
+  β,
+)
+  return mul!(parent(CM), BM', AM', α, β)
```

Comment:
```suggestion
  mul!(parent(CM), BM', AM', α, β)
```

Reply:
Also personally I would prefer:
```julia
return mul!(CM', BM', AM', α, β)
```
and for the `Transpose` version:
```julia
  mul!(transpose(CM), transpose(BM), transpose(AM), α, β)
```
I think it makes it clearer what the math is.

Reply:
Makes sense to me, I made these changes